src/main/java/io/anserini/collection/CoreCollection.java

```diff
@@ -123,6 +125,13 @@ public Document(JsonNode json) {
                   "doi:" + json.get("doi").asText();
         } else if ("abstract".equals(e.getKey())) {
           this.contents = json.get("title").asText() + "\n" + json.get("abstract").asText();
+        } else if ("topics".equals(e.getKey())) {
```

Comment:
Follow the logic here, i think it'd be better if we could check whether a value of the field is an array or a string instead of specifying "topics" so that it will enable more array-like value in other fields to be searched.

Reply:
Changed the conditional to be for all ArrayNode fields

Reply:
Changed the conditional to be for all ArrayNode fields